packages/quantic/cypress/integration/result-list/result-list-expectations.ts

```diff
@@ -0,0 +1,54 @@
+import {InterceptAliases} from '../../page-objects/search';
+import {should} from '../common-selectors';
+import {EventExpectations} from '../event-expectations';
+import {ResultListSelector, ResultListSelectors} from './result-list-selectors';
+
+function resultListExpectations(selector: ResultListSelector) {
+  return {
+    displayPlaceholder: (display: boolean) => {
+      selector
+        .placeholder()
+        .should(display ? 'exist' : 'not.exist')
+        .logDetail(`${should(display)} display the placeholder`);
+    },
+    displayResults: (display: boolean) => {
+      selector
+        .results()
+        .should(display ? 'exist' : 'not.exist')
+        .logDetail(`${should(display)} display results`);
+    },
+    resultsEqual: (resultsAlias: string) => {
```

Comment:
Those are good points and I had tried that but I reverted because the tests didn't look as neat given we have to `cy.get().then(...` in the tests. But that's just a personal thing about the visual aspect of the code so your arguments are quite compelling. I saw you made a similar suggestion on another test PR. If the change was made there then for consistency I'll definitely do it here as well.

Reply:
There is nothing wrong with the actual code. I was simply suggesting a slightly different approach. But you are right that maybe it just moves some of the complexity.

I'm fine keeping the code as is.